### PR TITLE
Intdir review updates

### PR DESCRIPTION
**Pull request by gselander**
Addressing #389

### CODE REVIEW

**Comment by gselander** [comment]
Not yet addressing random padding and general IANA considerations.

File: draft-ietf-lake-edhoc.md [hunk 1]
```diff
@@ -914,21 +914,21 @@ EAD_1 = 1* ead
...
-   * If the Initiator previously received from the Responder an error message with error code 2 containing SUITES_R (see {{wrong-selected}}) which indicates cipher suites supported by the Responder, then the Initiator SHOULD select its most preferred supported cipher suite among those (bearing in mind that error messages are not authenticated and may be forged).
-   * The Initiator MUST NOT change the supported cipher suites and the order of preference in SUITES_I based on previous error messages.
+The processing steps are detailed below and in {{wrong-selected}}.
+
+The Initiator SHALL compose message_1 as follows:
 
+* Construct SUITES_I as an array of cipher suites supported by I in order of preference with the first cipher suite in network byte order being the most preferred by I, and the last being the one selected by I for this session. If the cipher suite most preferred by I is selected then SUITES_I contains only that cipher suite and is encoded as an int. All cipher suites, if any, preferred by I over the selected one MUST be included. (See also {{wrong-selected}}.)
```
Comment (emanjon) [inline draft-ietf-lake-edhoc.md:929]: network byte order? Isn't the first element of an array defined by CBOR?

File: draft-ietf-lake-edhoc.md [hunk 2]
```diff
@@ -1383,152 +1384,264 @@ The Initiator and the Responder are allowed to select the connection identifier
...
+If supported by the device, it is RECOMMENDED that at least the long-term private keys are stored in a Trusted Execution Environment (TEE, {{I-D.ietf-teep-architecture}}) and that sensitive operations using these keys are performed inside the TEE.  To achieve even higher security it is RECOMMENDED that additional operations such as ephemeral key generation, all computations of shared secrets, and storage of the PRK keys can be done inside the TEE. The use of a TEE aims at preventing code within that environment to be tampered with, and preventing data used by such code to be read or tampered with by code outside that environment.
```
Comment (emanjon) [inline draft-ietf-lake-edhoc.md:1387]: TEE is a lot bigger than IETF, and not really something IETF is doing. TEEP should probably just be an example.

File: draft-ietf-lake-edhoc.md [hunk 3]
```diff
@@ -1383,152 +1384,264 @@ The Initiator and the Responder are allowed to select the connection identifier
...
+| > 65535 | Private Use                  |                   |
++---------+------------------------------+-------------------+
+
+~~~~~~~~~~~
+{: #fig-exporter-label title="EDHOC Exporter Label"}
+
+
+~~~~~~~~~~~
+
++----------------+-------------------------------------+
```
Comment (emanjon) [inline draft-ietf-lake-edhoc.md:1432]: Lets have a discussion in LAKE regarding registration procedures.

File: draft-ietf-lake-edhoc.md (hunk 1 above)
Comment (gselander) [reply draft-ietf-lake-edhoc.md]: Changed to first ... in the array

File: draft-ietf-lake-edhoc.md (hunk 2 above)
Comment (gselander) [reply draft-ietf-lake-edhoc.md]: Made into an example

File: draft-ietf-lake-edhoc.md (hunk 3 above)
Comment (gselander) [reply draft-ietf-lake-edhoc.md]: Updated. More reviews are welcome.

**Comment by gselander** [comment]
Now addressed essentially all comments from intdir. No change is made on random padding since that feature was requested by the WG. The IANA section is quite changed so expect to see more comments on that.

**Comment by gselander** [comment]
The main uncommented part of this PR is the new content in the IANA section. Plan to merge soon.

File: draft-ietf-lake-edhoc.md [hunk 4]
```diff
@@ -1283,7 +1284,7 @@ Implementations MAY support message_4. Error codes (ERR_CODE) 1 and 2 MUST be su
 
 Implementations MUST support EAD.
 
-Implementations MUST support cipher suite 2 and 3. Cipher suites 2 (AES-CCM-16-64-128, SHA-256, 8, P-256, ES256, AES-CCM-16-64-128, SHA-256) and 3 (AES-CCM-16-128-128, SHA-256, 16, P-256, ES256, AES-CCM-16-64-128, SHA-256) only differ in the size of the MAC length, so supporting one or both of these is no essential difference. Implementations only need to implement the algorithms needed for their supported methods.
+Implementations MUST support cipher suite 2 and 3. Cipher suites 2 (AES-CCM-16-64-128, SHA-256, 8, P-256, ES256, AES-CCM-16-64-128, SHA-256) and 3 (AES-CCM-16-128-128, SHA-256, 16, P-256, ES256, AES-CCM-16-64-128, SHA-256) only differ in the size of the MAC length, so supporting one or both of these is not significantly different. Implementations only need to implement the algorithms needed for their supported methods.
```
Comment (emanjon) [inline draft-ietf-lake-edhoc.md:1287]: >supporting one or both of these is not significantly different

?

The original text wanted to say that is was as easy to support 1 as 2.

**Review by emanjon — APPROVED** [review]

**Pull request merged**